pkg/ruler/ruler_test.go

```diff
@@ -844,9 +899,90 @@ func TestGetRules(t *testing.T) {
 				time.Sleep(100 * time.Millisecond)
 			}
 
+			scrape := func(rID string, scrapeCh chan prometheus.Metric, doneCh chan struct{}) {
```

Comment:
If unit test is hard to add let's just add it to the integration test?

Reply:
Yes, I will add an integration test